happi/cli.py

```diff
@@ -94,53 +96,68 @@ def search(ctx, show_json, names, search_criteria):
                 'Received duplicate search criteria %s=%r (was %r)',
                 criteria, value, client_args[criteria]
             )
-            return
-        if value.replace('.', '').isnumeric():
-            logger.debug('Changed %s to float', value)
-            value = str(float(value))
+            raise click.Abort()
 
         if is_a_range(value):
             start, stop = value.split(',')
             start = float(start)
             stop = float(stop)
-            is_range = True
             if start < stop:
-                range_list = client.search_range(criteria, start, stop)
+                new_range_list = client.search_range(criteria, start, stop)
             else:
                 logger.error('Invalid range, make sure start < stop')
+                raise click.Abort()
+
+            if not range_found:
+                # if first range, just replace
+                range_found = True
+                range_list = new_range_list
+            else:
+                # subsequent ranges, only take intersection
+                final_range = [item for item in new_range_list
+                               if item in range_list]
+                range_list = final_range
+
+            if not range_list:
+                # we have searched via a range query.  At this point
+                # no matches, or intesection is empty. abort early
+                logger.error('No devices found')
+                return
+
+            continue
 
-        # skip the criteria for range values
-        # it won't be a valid criteria for search_regex()
-        if is_range:
-            pass
+        elif is_number(value):
+            logger.debug('Changed %s to float', value)
+            value = str(float(value))
```

Comment:
Time to bring this up again - in your digging, do you understand why we go from `str` -> `float` -> `str`?

Reply:
It's because these arguments get passed to `client.search_regex`, which requires strings.  I assume we went `str` -> `float` to make sure the number was formatted correctly, but as I think about it more deeply, I wonder if this works for EntryItems that require `int`s.  I'll look into this some more.

Reply:
Ah yes if we have some value that's stored as an int rather than a float, the search_regex method won't work at all.  I'm willing to bet we've never run into this because our only numeric field is the float field `z`

Reply:
For now I'll leave this in.  For our current use case, it allows users to search `z=735` and still match `z=735.0`.  It's something that will need to be looked at in the future, however.

Reply:
Sounds good. I'm glad you dug into those details - and it's worth an issue/follow-up at some point for sure.